packages/@aws-cdk/aws-ecs-patterns/lib/base/application-load-balanced-service-base.ts

```diff
@@ -313,21 +322,25 @@ export abstract class ApplicationLoadBalancedServiceBase extends cdk.Construct {
     if (props.certificate !== undefined && props.protocol !== undefined && props.protocol !== ApplicationProtocol.HTTPS) {
       throw new Error('The HTTPS protocol must be used when a certificate is given');
     }
+
     const protocol = props.protocol !== undefined ? props.protocol :
       (props.certificate ? ApplicationProtocol.HTTPS : ApplicationProtocol.HTTP);
 
+    const containerProtocol = props.containerProtocol !== undefined ? props.containerProtocol : ApplicationProtocol.HTTP;
+
     const targetProps = {
-      port: 80
+      port: 80,
+      protocol: containerProtocol,
     };
 
     this.listener = loadBalancer.addListener('PublicListener', {
       protocol,
       port: props.listenerPort,
       open: true
     });
-    this.targetGroup = this.listener.addTargets('ECS', targetProps);
 
-    if (protocol === ApplicationProtocol.HTTPS) {
+    this.targetGroup = this.listener.addTargets('ECS', targetProps);
+    if (protocol === ApplicationProtocol.HTTPS || containerProtocol === ApplicationProtocol.HTTPS) {
```

Comment:
Although it would be a bit odd, having the LB serve 80 when the container serves HTTPS wouldn't require the LB to have a domain name I think?